lib/rouster.rb

```diff
@@ -292,60 +293,87 @@ def inspect
   #
   # runs a command inside the Vagrant VM
   #
-  # returns output (STDOUT and STDERR) from command run, sets @exitcode
-  # currently determines exitcode by tacking a 'echo $?' onto the command being run, which is then parsed out before returning
+  # returns output (STDOUT and STDERR) from command run, pushes to self.ssh_exitcode
   #
   # parameters
   # * <command> - the command to run (sudo will be prepended if specified in object instantiation)
   # * [expected_exitcode] - allows for non-0 exit codes to be returned without requiring exception handling
-  def run(command, expected_exitcode=[0])
+  def run( command, expected_exitcode = 0, sudo = self.uses_sudo? )
+
+    cmd = {
+      :command           => command,
+      :sudo              => sudo,
+      :stdout            => String.new,
+      :stderr            => String.new,
+      :expected_exitcode => Array( expected_exitcode ),
+      :exitcode          => nil,
+    }
+
+    cmd[:final_command] = cmd[:sudo] ? sprintf( 'sudo bash -c "%s"', command ) : command
```

Comment:
any reason not to do this in the hash definition itself? think this is a holdover from before we made `sudo` a parameter